creator-node/tsconfig.build.json

```diff
@@ -0,0 +1,16 @@
+// Config used to build the app (no tests)
```

Comment:
mind documenting the effective diff btwn old `tsconfig.json` and this new file?

Reply:
also if there is a non-zero diff - other than just refactoring - mind documenting this in PR description since its a critical path change to every CN build

Reply:
good call -- added to PR description. tldr of the effective diff:
- main config used when building app doesn't worry about tests
- separate config knows how to handle tests. this is used by VSCode for error-finding + used by testing command (`npm run test`)
- proper type recognition for mocha and node
- tests put into their own output directory (`build_test`) to avoid cluttering the app build dir (`build`)